File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    tmp = []
    tmp.append(words[0])
    for i in range(1, len(words)):
        if words[i] in tmp:
            break
        else:
            if words[i][0] != words[i-1][-1]:
                break
        tmp.append(words[i])
    
    return [0, 0] if len(tmp) == len(words) else [(i % n) + 1, (i // n) + 1]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.swift
import Foundation

func solution(_ n:Int, _ words:[String]) -> [Int] {
    var usedWords: [String] = []
    var lastAlphabet = words[0].first
    for i in 0..<words.count {
        if usedWords.contains(words[i]) || lastAlphabet != words[i].first {
            return [i%n+1, i/n+1]
        }
        usedWords.append(words[i])
        lastAlphabet = words[i].last
    }
    return [0, 0]
}